pkg/watchdog/watchdog.go

```diff
@@ -0,0 +1,236 @@
+		return wi, err
+	}
+	return wi, nil
+}
+
+// GetStatus returns the current status.
+func (w *Watchdog) GetStatus() (int32, error) {
+	var flags int32
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocGetStatus, uintptr(unsafe.Pointer(&flags))); err != 0 {
+		return StatusUnknown, err
+	}
+	return flags, nil
+}
+
+// GetBootStatus returns the status at the last reboot.
+func (w *Watchdog) GetBootStatus() (int32, error) {
+	var flags int32
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocGetBootStatus, uintptr(unsafe.Pointer(&flags))); err != 0 {
+		return StatusUnknown, err
+	}
+	return flags, nil
+}
+
+// GetTemp returns temperature in degrees Fahrenheit.
+func (w *Watchdog) GetTemp() (int32, error) {
+	var temp int32
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocGetTemp, uintptr(unsafe.Pointer(&temp))); err != 0 {
+		return 0, err
+	}
+	return temp, nil
+}
+
+// SetOptions can be used to control some aspects of the cards operation.
+func (w *Watchdog) SetOptions(options int32) error {
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocSetOptions, uintptr(unsafe.Pointer(&options))); err != 0 {
+		return err
+	}
+	return nil
+}
+
+// KeepAlive pets the watchdog.
+func (w *Watchdog) KeepAlive() error {
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocKeepAlive, 0); err != 0 {
+		return err
+	}
+	return nil
+}
+
+// SetTimeout sets the watchdog timeout on the fly. It returns the real timeout
+// used and this timeout might differe from the requested one due to limitation
+// of the hardware.
+func (w *Watchdog) SetTimeout(timeout int32) (int32, error) {
+	originalTimeout := timeout
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocSetTimeout, uintptr(unsafe.Pointer(&timeout))); err != 0 {
+		return 0, err
+	}
+	if originalTimeout != timeout {
+		log.Printf("Watchdog timeout set to %ds, wanted %ds", timeout, originalTimeout)
+	}
+	return timeout, nil
+}
+
+// GetTimeout returns the current watchdog timeout.
+func (w *Watchdog) GetTimeout() (int32, error) {
+	var timeout int32
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocGetTimeout, uintptr(unsafe.Pointer(&timeout))); err != 0 {
+		return 0, err
+	}
+	return timeout, nil
+}
+
+// SetPreTimeout sets the watchdog pretimeout on the fly. The pretimeout is the
+// number of seconds before the timeout before triggering the preaction (such
+// as an NMI, interrupt, ...)
+func (w *Watchdog) SetPreTimeout(timeout int32) (int32, error) {
+	originalTimeout := timeout
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, w.f.Fd(), wdiocSetPreTimeout, uintptr(unsafe.Pointer(&timeout))); err != 0 {
+		return 0, err
+	}
+	if originalTimeout != timeout {
+		log.Printf("Watchdog pretimeout set to %ds, wanted %ds", timeout, originalTimeout)
+	}
+	return timeout, nil
+}
+
+// GetPreTimeout returns the current watchdog pretimeout.
+func (w *Watchdog) GetPreTimeout() (int32, error) {
```

Comment:
convert to time.Duration

Reply:
It can only ever be a multiple on seconds. Should the function round up, round down, round to nearest? Why not remove the element of surprise from the API and constrain it as an int?

The other problem is there are 2 sentinal values: -1 means the value should not be changed and 0 means there is no timeout.